lib/ecto/query/builder/select.ex

```diff
@@ -210,17 +214,17 @@ defmodule Ecto.Query.Builder.Select do
 
   def build(kind, query, binding, expr, env) do
     {query, binding} = Builder.escape_binding(query, binding, env)
-    {expr, {params, {take, subqueries}}} = escape(expr, binding, env)
+    {expr, {params, acc}} = escape(expr, binding, env)
     params = Builder.escape_params(params)
-    take   = {:%{}, [], Map.to_list(take)}
+    take = {:%{}, [], Map.to_list(Map.get(acc, :take, %{}))}
```

Comment:
Let's to the same with `take` and make sure it is always set to an empty map. Then we can do `acc.take` here and slightly simplify `add_take`. :)